draft-colitti-ipsecme-esp-ping.mkd

```diff
@@ -74,8 +75,74 @@ An IPv6 node that desires to determine whether the path to a particular destinat
 
 If the destination supports ESP, and wishes to reveal to the sender that it does so, it SHOULD reply with an ESP Echo Reply packet. ESP Echo Reply packets are ESP packets with an SPI value of (8-TBD), a Next Header value of 59, and no payload.
 
+The ESP Echo Request and Reply packets utilize the standard ESP packet format as described in Section 2 of {{!RFC4303}} with the following changes:
+
+* SPI set to
+  * [ESP-ECHO-REQUEST] for ESP Echo Request
+  * [ESP-ECHO-REPLY] for ESP Echo Reply
+* The Next Header field of the ESP header SHOULD be set to 59 (No Next Header).
+* No Integrity Check Value-ICV.
+
+The payload has the following format:
+
+     0                   1                   2                   3
+      0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+     |        ECHO Identifier        |      ECHO Sequence Number     |
+     +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+     |     Data ...                                                  |
+     +-+-+-+-+-
+
+
+    * ECHO Identifier: An identifier to aid in matching Echo Replies to Echo Requests. May be zero.
+
+    * ECHO Sequence Number: An identifier to aid in matching Echo Replies to Echo Requests. May be zero.
+
+    * Data: Zero or more octets of arbitrary data.
+
+
+Figure 1: ESP Echo Request and Reply Payload Overview
+
+An IPsec peer, prior to an IKE negotiation or after completing an IPSec negotiation, intending to ascertain the path's capability to support ESP packets to a specific destination, MAY send an ESP Echo Request packet to such destination.
+Should the destination support ESP and intend to communicate this capability to the potential IPsec peer, it SHOULD respond with an ESP Echo Reply packet.
+
+The sender MAY send ESP Echo packets with zero data. 
+When responding to an ESP Echo packet, the node SHOULD generate a packet devoid of any data.
+If the ESP Echo Request packet contain non-zero data, the resulting ESP Echo Reply packet size MUST NOT exceed the size of the ESP Echo Request packet received.
+
+# Discovering ESP Echo Support
+
+If no response is received to an ESP Echo Request packet, it can be caused by one of the following:
+
+* the peer doesn't support ESP Echo protocol.
+* there is no end-to-end ESP connectivity.
+
+Without some prior knowledge about ESP Echo support by the remote side, the sender can not distibguish those two scenarios.
+Therefore the sender SHOULD NOT treat lack of response as an indicator of end-to-end connectivity issues until an explicit confirmation of ESP Echo support by the peer is received. 
+The IPSec peer SHOULD announce ESP Echo support using IKE Notify Payload (more text to be added heve in the next version).
```

Comment:
```suggestion
The IPsec peer SHOULD announce ESP Echo support using IKE Notify Payload (more text to be added heve in the next version).
```
I am not in favour of any link to IKE.

Reply:
>I am not in favour of any link to IKE.

Would you be willing to elaborate? It was suggested by someone on the list and I think it makes sense, so if we are not willing to do this, we'd need a reason...